### PR TITLE
Multi-level ToC Implementation

### PR DESCRIPTION
**Pull request by dblotsky**
- Added support for nested ToCs using the `children` attribute
- Retroactively made past ToC files use relative links
- Made all dev ToCs use the new documentation information architecture

### CODE REVIEW

File: www/_data/toc/de-dev-manual.yml [hunk 1]
```diff
@@ -1,19 +1,70 @@
...
+        -   name: "Tooling Architecture"
+            url:
+-   name: "Create apps"
+    children:
+        -   name: "Create your first app"
+            url:
+        -   name: "Develop for platforms"
+            url: "guide/platforms/index.html"
+        -   name: "Manage versions and platforms"
+            url:
```
Comment (rakatyal) [inline www/_data/toc/de-dev-manual.yml]: This article is present in current TOC. platform_plugin_versioning_ref/index.html

**Comment by nikhilkh** [comment]
If an article does not have a URL do we make sure it does not show up in the displayed page?

**Comment by nikhilkh** [comment]
Looks like there are really two changes - use of relative links & multi-level TOC. It's hard to follow but I'm assuming only newer versions of the docs do get multi-level TOC - older versions remain unchanged.

**Comment by dblotsky** [comment]
@nikhilkh I made articles not show up if they don't have a URL. As for older versions, they were only updated to use relative links.

File: www/_data/toc/de-dev-manual.yml (hunk 1 above)
Comment (dblotsky) [inline www/_data/toc/de-dev-manual.yml]: Got it. Thanks!

**Comment by nikhilkh** [comment]
This looks great! One issue: "Architecture" has no children with URLs and it shows up in the TOC. 
image

**Comment by dblotsky** [comment]
@nikhilkh that is a difficult bug to fix because it relies on knowing that all children of "Architecture" also don't have any URIs. It will start to have some pages under it very soon, so I think it's ok to have the title be empty in the short term.

**Comment by nikhilkh** [comment]
Could you comment "Architecture" and it's sub-sections? We might not get to write these topics till April or something. It's best to not to have empty non-clickable placeholders in TOC.

**Comment by dblotsky** [comment]
@nikhilkh Fixed. Merging.
